src/main/java/de/kekru/struktogrammeditor/control/GlobalSettings.java

```
@@ -236,50 +226,61 @@ private static void loadSettings(){
 			for(int i=0; i < EinstellungsDialog.anzahlStruktogrammElemente; i++){
 				s = pr.getProperty("caption"+i);//Startbeschriftung der Struktogrammelemente
 				if(s != null){
-					elementBeschriftungenZumEinfuegenInDasStruktogramm[i] = s;
+					elementBeschriftungenZumEinfuegenInDasStruktogramm.set(i, s);
 				}				 	
 			}		
-		}		
+		} else {
+			try {
+				einstellungsDatei.createNewFile();
+			} catch (IOException e) {
+				System.err.println("Could not create Settingsfile!");
+				System.err.println(e.getMessage());

```

Best practice would be to throw a RuntimeException, catch it somewhere and then show an OptionPane with the exceptions message to the user.  

But its okay, since it is not done in the code on the other places

---

Why throw an Exception and catch it later? We could just show the message and return

---

For "real" projects I would prefer a layered architecture.  
Something like:

UI Layer -> Services-Layer -> Filesystem-Things

This GlobalSettings.java would be anywhere in the Services-Layer and therefore shuld not know about any UI implementations (e.g. JOptionPane). So it should only throw an Exception.  
The exception can be caught in the UI-Layer an could be presented to the User.  

On this way you could separate the UI for an Error-Message from the service, where the message was created and you could implement the JOptionPane once in the JOptionPane.  

Thats the theory, but this is the struktogrammeditor, we don't really need to such an architecture here

But maybe you could throw an Exception here and catch it in the GlobalSettings.init() and show the JOptionPane there. Then would would catch all exceptions, that could appear anywhere in this class